custom_components/moonraker/sensor.py

```diff
@@ -201,11 +229,41 @@ class MoonrakerSensorDescription(SensorEntityDescription):
 async def async_setup_entry(hass, entry, async_add_entities):
     """Setup sensor platform."""
     coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities([MoonrakerSensor(coordinator, entry, desc) for desc in SENSORS])
+
+    object_list = await coordinator.async_fetch_data(METHOD.PRINTER_OBJECTS_LIST)
+    opt_sensors = await async_setup_optional_sensors(object_list)
```

Comment:
Nit pick, I have seen this being done directly in the `async_setup_entry` not in a sub function.

in other word, the for loop directly here. 

No strong feeling, 
The function is nice because it is easier to read, the no-function option look to be more Home-Assistant way of doing it

Reply:
I'll keep it that way, function are nice :)